docs/how-to/partitions.rst

```diff
@@ -0,0 +1,136 @@
+******************************************
+Add partition support to an application
+******************************************
+
+Partitions basics
+=================
+
+In the below examples, we work with three partitions: ``default``, ``kernel``, and ``bar-baz``.  The ``default`` and ``kernel`` partitions do not have a namespace.  The ``bar-baz`` partition is part of the ``component`` namespace.
+
+``default`` must always be the first listed partition.
+
+Partition names and namespace names must consist of *only* lower-case alphabetic characters, unless a partition exists under a namespace, in which case it may also contain hyphen characters, though the first and last characters must still be alphabetic.
+
+.. _app_changes:
+
+Required application changes
+============================
+
+To add partition support to an application, two basic changes are needed to the ``craft-application``.
+
+#. Enable the feature
+
+   When creating the :class:`AppMetadata <craft_application.AppMetadata>`, specify that the application will use the partitions feature:
+
+   .. code-block:: python
+
+     from craft_application import AppMetadata, AppFeatures
```

Comment:
As @mr-cal mentioned above, craft-parts is not aware of the existence of craft-application. If in craft-parts, this documentation should use only the craft-parts API. Craft-application usage should be part of the craft-application documentation.